src/main/java/org/gridsuite/study/server/networkmodificationtree/AbstractNodeRepositoryProxy.java

```diff
@@ -77,4 +81,41 @@ public void updateNode(AbstractNode node) {
     public void deleteAll(Set<UUID> collect) {
         nodeInfoRepository.deleteByIdNodeIn(collect);
     }
+
+    public Optional<String> getVariantId(UUID nodeUuid, boolean generateId) {
+        AbstractNode node = getNode(nodeUuid);
```

Comment:
Make different implementations in subclasses ?

Reply:
Done